src/components/RecipesRecommended.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getXFirstElementsFromArray } from '../helpers/utils';
+import '../index.css';
+
+const DECREMENT = -1;
+const INCREMENT = 1;
+const INDEX = 5;
+
+class RecipesRecommended extends React.Component {
+  constructor({ recipesRecommended }) {
+    super({ recipesRecommended });
+
+    this.state = {
+      slideIndex: 0,
+      // twoRecommended: getXFirstElementsFromArray(recipesRecommended, QUANTITY),
+    };
+
+    this.getCarouselElements = this.getCarouselElements.bind(this);
+    this.renderCarouselElements = this.renderCarouselElements.bind(this);
+  }
+
+  componentDidMount() {
+    const { recipesRecommended } = this.props;
+    console.log('valor de recipesRecommended em didUpdate');
+    console.log(recipesRecommended);
+  }
+
+  componentDidUpdate() {
+    const { slideIndex } = this.state;
+    console.log('slideIndex em didUpdate');
+    console.log(slideIndex);
+  }
+
+  getCarouselElements() {
+    const { recipesRecommended } = this.props;
+    const { slideIndex } = this.state;
+
+    return getXFirstElementsFromArray(recipesRecommended, INDEX + 1)
+      .map((item, index) => (
+        <div
+          data-testid={ `${index}-recomendation-card` }
+          className="card"
+          // className={ (!(index === slideIndex || index === slideIndex + 1)
+          //   && !(slideIndex === INDEX && index === 0)) ? 'displayNone card' : 'card' }
+          key={ index }
+          hidden={ !(index === slideIndex || index === slideIndex + 1)
+            && !(slideIndex === INDEX && index === 0) }
+        >
+          <div className="numbertext">
+            {`${index + 1} / 6`}
+          </div>
+          <img
+            src={ item.strMealThumb || item.strDrinkThumb }
+            style={ { width: '100%' } }
+            alt="Recomendadação"
+          />
+          <div
+            data-testid={ `${index}-recomendation-title` }
+            className="text"
+          >
+            { item.strMeal || item.strDrink }
+          </div>
+        </div>
+      ));
+  }
+
+  showSlides(n) {
+    // const { recipesRecommended } = this.props;
```

Comment:
O que acha de remover esses comentários? 😉 
```suggestion
    return getXFirstElementsFromArray(recipesRecommended, INDEX + 1)
      .map((item, index) => (
        <div
          data-testid={ `${index}-recomendation-card` }
          className="card"
          key={ index }
          hidden={ !(index === slideIndex || index === slideIndex + 1)
            && !(slideIndex === INDEX && index === 0) }
        >
          <div className="numbertext">
            {`${index + 1} / 6`}
          </div>
          <img
            src={ item.strMealThumb || item.strDrinkThumb }
            style={ { width: '100%' } }
            alt="Recomendadação"
          />
          <div
            data-testid={ `${index}-recomendation-title` }
            className="text"
          >
            { item.strMeal || item.strDrink }
          </div>
        </div>
      ));
  }

  showSlides(n) {
```